core/factory.cfc

```diff
@@ -4,12 +4,19 @@
 		this.beans = structNew();
 		this.transients = structNew();
 		//functionality
-		function init(externalBeanFactory){
-			if (structKeyExists(arguments, "externalBeanFactory")) {
+	</cfscript>
+
+	<cffunction name="init" output="false">
+		<cfargument name="externalBeanFactory" default="">
+	<cfscript>
+			if (structKeyExists(arguments, "externalBeanFactory") AND arguments.externalBeanFactory neq "") {
```

Comment:
I'm curious: Would changing this from:

```js
function init(externalBeanFactory){
```

to:

```js
function init(externalBeanFactory = ""){
```

... also fix it? That would seem to be the better fix IMO if it works.

Reply:
Nope, I have already tried this. it was throwing syntax error as cfml parsing in cf 8

Reply:
Great, thanks for trying it!